### DISCUSSION TITLE
Unable to load members in object browser

### DISCUSSION THREAD

**Discussion by figgie** [codefori/vscode-ibmi#1885] [IBM i]
Hi, I'm not able to view any objects inside the object browser. The files load but none of their members are visible. On trying to expand the file, I see the following error in the Output section: 
```
With MEMBERS As (
        SELECT
          rtrim(cast(a.system_table_schema as char(10) for bit data)) as LIBRARY,
          b.avgrowsize as RECORD_LENGTH,
          a.iasp_number as ASP,
          rtrim(cast(a.system_table_name as char(10) for bit data)) AS SOURCE_FILE,
          rtrim(cast(b.system_table_member as char(10) for bit data)) as NAME,
          coalesce(rtrim(cast(b.source_type as varchar(10) for bit data)), '') as TYPE,
          coalesce(rtrim(varchar(b.partition_text)), '') as TEXT,
          b.NUMBER_ROWS as LINES,
          extract(epoch from (b.CREATE_TIMESTAMP))*1000 as CREATED,
          extract(epoch from (b.LAST_SOURCE_UPDATE_TIMESTAMP))*1000 as CHANGED
        FROM qsys2.systables AS a
          JOIN qsys2.syspartitionstat AS b
            ON b.table_schema = a.table_schema AND
              b.table_name = a.table_name
      )
      Select * From MEMBERS
      Where LIBRARY = '<library>'
        And SOURCE_FILE = '<file>'
        And NAME Like '%'
        And TYPE Like '%'
      Order By NAME ASC
{
    "code": 4,
    "signal": null,
    "stdout": "DB2>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n  ?>\n\n **** CLI ERROR *****\n         SQLSTATE: 42601\nNATIVE ERROR CODE: -104\nToken EPOCH var ikke gyldig. Gyldige tokens: DAY HOUR YEAR MONTH MINUTE SECOND.",
    "stderr": ""
}
```

The error translates to `Token EPOCH was not valid. Valid tokens: DAY HOUR YEAR MONTH MINUTE SECOND`.

I did a bit of digging around and saw that the `extract` function accepts `epoch` only in non z/OS Db2 and also only in versions >= 11.1.

[Z/OS EXTRACT Documentation](https://www.ibm.com/docs/en/db2-for-zos/11?topic=functions-extract)
[Non-Z/OS Extract Documentation](https://www.ibm.com/docs/en/db2/11.1?topic=functions-extract)

I can't seem to find any mention of these requirements in the requirements section at the [ReadMe](https://codefori.github.io/docs/#/./README). If this is needed can we modify the requirements or could you suggest any workaround for the same? 

**Expected behavior**
I should be able to see members of the files.

**Environment (please complete the following information):**
 - Extension version [e.g. 0.x.x] : v2.7.0
 - IBM i OS version [e.g. 7.x] : V7R1M0

**Comment by chrjorgensen** [comment]
@figgie The [EPOCH](https://www.ibm.com/docs/en/i/7.2?topic=functions-extract) date value in [EXTRACT](https://www.ibm.com/docs/en/i/7.2?topic=functions-extract) arrived with IBM i 7.2 back in 2014... we do not support any versions of IBM i before 7.3, which became available in 2016. 😢 

IBM i 7.1 went out of support in 2018, so the server is really in need of an upgrade. 😍 

The IBM i release life cycle can be seen here: [https://www.ibm.com/support/pages/release-life-cycle](https://www.ibm.com/support/pages/release-life-cycle)

**Reply by johnbouris** [reply]
I have the exact same problem as figgie, but I am on i5/OS V7.3.  Any thoughts?
I would love to get this working, but no members will list below source files.

**Reply by worksofliam** [reply]
@johnbouris ensure you have the latest db2 PTFs.

**Reply by johnbouris** [reply]
Thanks.  That makes sense.  I downloaded the cume and put it in an image catalog... Just gotta bite the bullet.

**Reply by e1mais** [reply]
Hi, @worksofliam 

We're also having this same issue, but it seems like it's a different reason. The name that my org uses for Source files starts with the "Ñ" character and the extension SQL's is using # instead.

Any advice for the solve of the issue would be very appreciated.

PSD: I can't provide images because we have restrictions to use GitHub :(

Greatings 

IBM: V7R4M0 
Extension version: v2.13.0

**Reply by chrjorgensen** [reply]
@e1mais Seems like the same problem as in issue #2239 - maybe following this issue will make you find the solution?
